tools/tools.go

```diff
@@ -0,0 +1,178 @@
+package tools
+
+import (
+	"errors"
+	"strings"
+	gotmpl "text/template"
+
+	"github.com/ollama/ollama/api"
+	"github.com/ollama/ollama/template"
+)
+
+// Sentinel errors for parsing states
+var (
+	ErrPartialPrefix = errors.New("partial prefix detected")
+
+	ErrPrefixNotFound = errors.New("prefix not found")
+
+	ErrInvalidToolCall = errors.New("invalid tool call format")
+
+	ErrAccumulateMore = errors.New("need to accumulate more content")
+)
+
+type Parser struct {
+	greedyParse bool
+	prefixFound bool
+	tmpl        gotmpl.Template
+	sb          strings.Builder
+	prefix      string
+	index       int
+	name        string
+	arguments   string
+	done        bool
+}
+
+// checkPrefix processes a string to find and handle a prefix pattern.
+//
+// Returns:
+//   - The processed string with prefix removed if found
+//   - error: ErrPartialPrefix if prefix is incomplete, ErrPrefixNotFound if not found, or nil if successful
+func (p *Parser) checkPrefix(s string) (string, error) {
+	// Keep original for overlap checks
+	original := s
+
+	if s == "" {
+		return "", nil
+	}
+
+	// If no prefix defined, just return trimmed string
+	if p.prefix == "" {
+		return s, nil
+	}
+
+	// Check for prefix at start of string
+	if processedStr, hasPrefix := strings.CutPrefix(s, p.prefix); hasPrefix {
+		// Found prefix at start - accumulate for potential tool
+		p.prefixFound = true
+		return processedStr, nil
+	}
+
+	// Check if prefix overlaps end of string
+	if overlap := suffixOverlap(original, p.prefix); overlap > 0 {
+		// Return everything except overlapping portion
+		p.sb.Reset()
+		p.sb.WriteString(original[len(original)-overlap:])
+		return original[0 : len(original)-overlap], ErrAccumulateMore
```

Comment:
As discussed above let's remove this error and instead return nothing

Reply:
This is just for internal state management - not returned to user